rio/src/main/java/org/semanticweb/owlapi/rio/utils/OWLAPISimpleSAXParser.java

```diff
@@ -0,0 +1,358 @@
+	 * <tt>XMLReader</tt> using <tt>info.aduna.xml.XMLReaderFactory</tt> for
+	 * parsing the XML. One must set a <tt>SimpleSAXListener</tt> on this
+	 * object before calling one of the <tt>parse()</tt> methods.
+	 * 
+	 * @throws SAXException
+	 *         If the OWLAPISimpleSAXParser was unable to create an XMLReader.
+	 * 
+	 * @see #setListener
+	 * @see XMLReader
+	 * @see XMLReaderFactory
+	 */
+	public OWLAPISimpleSAXParser()
+		throws SAXException
+	{
+		this(XMLReaderFactory.createXMLReader());
+	}
+
+	/*---------*
+	 * Methods *
+	 *---------*/
+
+	/**
+	 * Sets the (new) listener that should receive any events from this parser.
+	 * This listener will replace any previously set listener.
+	 * 
+	 * @param listener
+	 *        The (new) listener for events from this parser.
+	 */
+	public void setListener(SimpleSAXListener listener) {
+		this.listener = listener;
+	}
+
+	/**
+	 * Gets the listener that currently will receive any events from this parser.
+	 * 
+	 * @return The listener for events from this parser.
+	 */
+	public SimpleSAXListener getListener() {
+		return listener;
+	}
+
+	/**
+	 * Sets whether leading and trailing whitespace characters in text elements
+	 * should be preserved. Such whitespace characters are discarded by default.
+	 */
+	public void setPreserveWhitespace(boolean preserveWhitespace) {
+		this.preserveWhitespace = preserveWhitespace;
+	}
+
+	/**
+	 * Checks whether leading and trailing whitespace characters in text elements
+	 * are preserved. Defaults to <tt>false</tt>.
+	 */
+	public boolean isPreserveWhitespace() {
+		return preserveWhitespace;
+	}
+
+	/**
+	 * Parses the content of the supplied <tt>File</tt> as XML.
+	 * 
+	 * @param file
+	 *        The file containing the XML to parse.
+	 */
+	public void parse(File file)
+		throws SAXException, IOException
+	{
+		InputStream in = new FileInputStream(file);
+		try {
+			parse(in);
+		}
+		finally {
+			try {
+				in.close();
+			}
+			catch (IOException ignore) {
+			}
+		}
+	}
+
+	/**
+	 * Parses the content of the supplied <tt>InputStream</tt> as XML.
+	 * 
+	 * @param in
+	 *        An <tt>InputStream</tt> containing XML data.
+	 */
+	public void parse(InputStream in)
+		throws SAXException, IOException
+	{
+		parse(new InputSource(in));
+	}
+
+	/**
+	 * Parses the content of the supplied <tt>Reader</tt> as XML.
+	 * 
+	 * @param reader
+	 *        A <tt>Reader</tt> containing XML data.
+	 */
+	public void parse(Reader reader)
+		throws SAXException, IOException
+	{
+		parse(new InputSource(reader));
+	}
+
+	/**
+	 * Parses the content of the supplied <tt>InputSource</tt> as XML.
+	 * 
+	 * @param inputSource
+	 *        An <tt>InputSource</tt> containing XML data.
+	 */
+	private synchronized void parse(InputSource inputSource)
+		throws SAXException, IOException
+	{
+		SimpleSAXDefaultHandler handler = new SimpleSAXDefaultHandler();
+		xmlReader.setContentHandler(handler);
+		xmlReader.setErrorHandler(handler);
```

Comment:
The default implementation of [DefaultHandler::fatalError](http://www.saxproject.org/apidoc/org/xml/sax/helpers/DefaultHandler.html#fatalError(org.xml.sax.SAXParseException) rethrows the SaxParseException. It's when this re-thrown error is caught in the TrixParser that the  error listener is invoked (this fooled me at first). 

Also, isn't it great how javax.xml.parsers.SAXParser requires DefaultHandler (as an argument to parse, overriding any handlers set on the underlying parser), but javax.xml.parsers.SAXParser requires individual handlers to be set on the XMLReader.